src/helpers/task-update.ts

```diff
@@ -3,42 +3,43 @@ import { Context } from "../types/context";
 import { ListForOrg, ListIssueForRepo } from "../types/github-types";
 import { remindAssigneesForIssue, unassignUserFromIssue } from "./remind-and-remove";
 import { getDeadlineWithThreshold } from "./task-deadline";
-import { getTaskMetadata } from "./task-metadata";
+import { getTaskAssignmentDetails } from "./task-metadata";
 
 export async function updateTaskReminder(context: Context, repo: ListForOrg["data"][0], issue: ListIssueForRepo) {
   const { logger } = context;
 
-  let metadata, lastCheck, deadlineWithThreshold, reminderWithThreshold, now;
+  let deadlineWithThreshold, reminderWithThreshold;
+  const now = DateTime.now();
 
-  const handledMetadata = await getTaskMetadata(context, repo, issue);
+  const handledMetadata = await getTaskAssignmentDetails(context, repo, issue);
 
   if (handledMetadata) {
-    metadata = handledMetadata.metadata;
-    lastCheck = handledMetadata.lastCheck;
-
-    const handledDeadline = await getDeadlineWithThreshold(context, metadata, issue, lastCheck);
+    const handledDeadline = await getDeadlineWithThreshold(context, handledMetadata, issue);
     if (handledDeadline) {
       deadlineWithThreshold = handledDeadline.deadlineWithThreshold;
       reminderWithThreshold = handledDeadline.reminderWithThreshold;
-      now = handledDeadline.now;
+
+      logger.info(`Handling metadata and deadline for ${issue.html_url}`, {
+        initialDeadline: DateTime.fromISO(handledMetadata.startPlusLabelDuration).toLocaleString(DateTime.DATETIME_MED),
+        now: now.toLocaleString(DateTime.DATETIME_MED),
+        reminderWithThreshold: reminderWithThreshold.toLocaleString(DateTime.DATETIME_MED),
+        deadlineWithThreshold: deadlineWithThreshold.toLocaleString(DateTime.DATETIME_MED),
+      });
     }
   }
 
-  if (!metadata || !lastCheck || !deadlineWithThreshold || !reminderWithThreshold || !now) {
+  if (!deadlineWithThreshold || !reminderWithThreshold) {
     logger.error(`Failed to handle metadata or deadline for ${issue.html_url}`);
     return false;
   }
 
   if (now >= deadlineWithThreshold) {
+    // if the issue is past due, we should unassign the user
```

Comment:
Ignore deadline. This is strictly unassigning based on config idle timeout

Reply:
`deadlineWithThreshold` here is either `lastActivity.createdAt + threshold` or `(issue.createdAt + labelDuration) + threshold` or `issue.created + threshold`. So this is unassigning based on the configurable idle timeout setting

Reply:
You should remove misleading comments.